test/tests/functional/pbs_hook_endjob.py

```diff
@@ -0,0 +1,1258 @@
+        stopping the MoM. Verify that the end job hook is executed.  Then
+        restart the mom and verify that the job is restarted and the end
+        job hook is executed again.
+        """
+        self.run_test_func(
+            self.endjob_delete_running_job_moms_stopped,
+            restart_moms=True,
+            job_rerunnable=True)
+
+    # FIXME: Possible server bug?  When a rerunnable job is deleted while the
+    # mom is stopped and restarted, the job is killed and requeued as expected.
+    # The job is rerun after the MoM is restarted and runs to completion;
+    # however, the job's substate indicates that it was deleted.
+    def test_hook_endjob_delete_running_single_job_as_user_rm(self):
+        """
+        Run a single rerunable job, but delete as root before completion after
+        stopping the MoM. Verify that the end job hook is executed.  Then
+        restart the mom and verify that the job is restarted and the end
+        job hook is executed again.
+        """
+        self.run_test_func(
+            self.endjob_delete_running_job_moms_stopped,
+            delete_user=TEST_USER,
+            restart_moms=True,
+            job_rerunnable=True)
+
+    def test_hook_endjob_force_delete_running_single_job_as_root_rm(self):
+        """
+        Run a single rerunable job, but force delete as root before completion
+        after stopping the MoM. Verify that the end job hook is executed.  Then
+        restart the mom and verify that the end job hook is not executed again.
+        """
+        self.run_test_func(
+            self.endjob_delete_running_job_moms_stopped,
+            delete_force=True,
+            restart_moms=True,
+            job_rerunnable=True)
+
+    def test_hook_endjob_force_delete_running_single_job_as_user_rm(self):
+        """
+        Run a single rerunable job, but force delete as user before completion
+        after stopping the MoM. Verify that the end job hook is executed.  Then
+        restart the mom and verify that the end job hook is not executed again.
+        """
+        self.run_test_func(
+            self.endjob_delete_running_job_moms_stopped,
+            delete_force=True,
+            delete_user=TEST_USER,
+            restart_moms=True,
+            job_rerunnable=True)
+
+    # FIXME: Possible bug in the server or qdel.  When an active job array with
+    # running subjobs is deleted while the MoM is stopped, deleting the job
+    # array appears to result in qdel returning success (exit code 0)
+    # immediately despite the server being unable to contact the mom and thus
+    # the deletion of the subjobs failing.  The subjobs are requeued and run
+    # again when resources are available, which seems to be the opposite of a
+    # successful delete.  Shouldn't qdel return a non-zero exit status since
+    # the delete was unsuccessful?
+    def test_hook_endjob_delete_running_array_job_as_root_sm(self):
```

Comment:
See previous FIXME comment.

Reply:
The failure scenario is described in issue #2486.  An `@skip` was added to the test in 0eaabbb7425b1a159e3dba3d2bd58c5976577ffd.